arcsi/api/show.py

```diff
@@ -389,30 +389,52 @@ def view_show_archive(show_slug):
     show = show_query.first()
     if show:
-        # subquery = session.query(Item.id).filter(blabla -timedelta(day=1)).all().subquery()
-        # query = session.query(Show).filter_by(blabla).(Item.id.in_(subquery))
         if show.cover_image_url:
             show.cover_image_url = do.download(
                 show.archive_lahmastore_base_url, show.cover_image_url
             )
         serial_show = show_archive_schema.dump(show)
-        show_items = [
-            show_item
-            for show_item in serial_show["items"]
-            if datetime.strptime(show_item.get("play_date"), "%Y-%m-%d")
-            + timedelta(days=1)
-            < datetime.today()
-        ]
-        serial_show["items"]=show_items
-        for item in serial_show["items"]:
-            item["image_url"] = do.download(
-                show.archive_lahmastore_base_url, item["image_url"]
-            )
-            item["name_slug"]=normalise(item["name"])
-        return serial_show
+        if (archived):
+            show_items = [
+                show_item
+                for show_item in serial_show["items"]
+                if (show_item.get("archived") &
+                    (datetime.strptime(show_item.get("play_date"), "%Y-%m-%d") + timedelta(days=1) < datetime.today()))
+            ]
+            if (latest):
+                show_items = sorted(show_items, key=lambda x: x["play_date"], reverse=True)
+                serial_show["items"] = show_items[0]
+                serial_show["items"]["image_url"] = do.download(show.archive_lahmastore_base_url, serial_show["items"]["image_url"])
+                serial_show["items"]["name_slug"] = normalise(serial_show["items"]["name"])
+                return serial_show
+            else:
+                serial_show["items"] = show_items
+                for item in serial_show["items"]:
+                    item["image_url"] = do.download(
+                        show.archive_lahmastore_base_url, item["image_url"]
+                    )
+                    item["name_slug"] = normalise(item["name"])
+                return serial_show
+        else:
+            if (latest):
+                serial_show["items"] = sorted(serial_show["items"], key=lambda x: x["play_date"], reverse=True)
+                serial_show["items"] = serial_show["items"][0]
+                serial_show["items"]["image_url"] = do.download(show.archive_lahmastore_base_url, serial_show["items"]["image_url"])
+                serial_show["items"]["name_slug"] = normalise(serial_show["items"]["name"])
+                return serial_show
+            else:
+                for item in serial_show["items"]:
+                    item["image_url"] = do.download(
+                        show.archive_lahmastore_base_url, item["image_url"]
+                    )
+                    item["name_slug"] = normalise(item["name"])
+                return serial_show
+
```

Comment:
This whole part is the type of pattern that exists currently elsewhere in the Api (I wrote a lot of it as well) but we should not replicate it instead trying to move away. 

The problem with this type of view construction is that is lazy in the bad way. 
1, It makes us do the work twice but also for the server we make it execute twice. 
2, Also in a bad way lazy for data structures. We reconstruct each db query, media downloads, python functions etc. multiple times. 
3, While it gives a shortcut when writing or thinking about it, it is hard to extend to other parameters, or change the relations, or the data we want to get. 

The nested `if-else` statements with `archived` and `latest` are a red alert for me in such situations, they alert that it's a pattern that will be dependent and coupled and that makes it easy to break, hard to maintain.